src/io/flutter/survey/FlutterSurveyNotifications.java

```diff
@@ -0,0 +1,162 @@
+/*
+ * Copyright 2019 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.survey;
+
+import com.intellij.ide.BrowserUtil;
+import com.intellij.ide.util.PropertiesComponent;
+import com.intellij.notification.Notification;
+import com.intellij.notification.NotificationType;
+import com.intellij.notification.Notifications;
+import com.intellij.openapi.actionSystem.AnAction;
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.fileEditor.FileEditorManager;
+import com.intellij.openapi.fileEditor.FileEditorManagerEvent;
+import com.intellij.openapi.fileEditor.FileEditorManagerListener;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.jetbrains.lang.dart.DartFileType;
+import icons.FlutterIcons;
+import io.flutter.FlutterInitializer;
+import io.flutter.FlutterMessages;
+import io.flutter.pub.PubRoot;
+import org.jetbrains.annotations.NotNull;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.TimeZone;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+// TODO(pq): remove for M39 / post: 2019/08/24 18:00 PST.
+public class FlutterSurveyNotifications {
+  private static final int NOTIFICATION_DELAY_IN_SECS = 3;
+
+  private static final String FLUTTER_LAST_SURVEY_CHECK_KEY = "FLUTTER_LAST_SURVEY_CHECK_KEY";
+  private static final long CHECK_INTERVAL_IN_MS = TimeUnit.HOURS.toMillis(40);
+
+  private static final String SURVEY_ACTION_TEXT = "Take Flutter Q3 Survey";
```

Comment:
`"Take Flutter Q3 Survey"` ==> `"Take Flutter's Q3 Survey"`? Not sure where or if this string is ever displayed to users

Reply:
I'm great with these text improvements but should defer to @jayoung-lee.  This would also diverge from VS Code...

Reply:
Does this get displayed to the users? I thought we are using "Help improve Flutter! Take our Q3 survey." for the pop up message.

Reply:
@jayoung-lee: you're correct.  The visible text is "Help improve Flutter! Take our Q3 survey."   (The other text is an internal label for the action.)

Reply:
EDIT: it does.  Trimmed to "Take survey"

![image](https://user-images.githubusercontent.com/67586/61814844-d269f000-adfd-11e9-9a4a-7c52632a7634.png)